### PR TITLE
Tech refresh, scala to 2.12.11 and a couple of other newer libraries …

### PR DESCRIPTION
**Pull request by rleibman**
…as well

# <INSERT TITLE HERE>

## Describe your change:

<INSERT DESCRIPTION HERE>

### Pull Request

* [ ] PR description is meaningful.
* [ ] There are no more than a few commits to merge.
* [ ] There are no more than a few lines of code to review.

### Code style

* [ ] Maximum line length does not require scrolling in Github
* [ ] FMT and ScalaStyle and ScalaFix are good

### Testing

* [ ] Unit tests for the added/modified code 
* [ ] Overall unit test coverage good
* [ ] Unit tests are passing

### Dependencies

* [ ] The plugin is added to itself
* [ ] The latest version is added to itself
* [ ] New dependencies do not have vulnerabilities

### Security

* [ ] There is no known vulnerability being added to this project
* [ ] Everything on this list is checked

### CODE REVIEW

File: build.sbt [hunk 1]
```diff
@@ -1,15 +1,17 @@
 soteriaAddSemantic := false
 
+scalaVersion in Global := "2.12.11"
```
Comment (leobenkel) [inline build.sbt:3]: Remove this. For SBT plugin, the scalaVersion will be the one corresponding to the one provided by the sbt version

**Comment by leobenkel** [comment]
Don't forget to raise `VERSION` number too

File: build.sbt (hunk 1 above)
Comment (rleibman) [reply build.sbt:3]: The problem is that for some stupid reason, the default version seems to be 2.12.7, so if I don't have this, I get:
```[error] java.lang.RuntimeException: [SoteriaPlugin] [SoteriaPlugin] Scala: 2.12.7 != [2.12.11 OR 2.11.12] !!!
```

Comment (leobenkel) [reply build.sbt:3]: There is no higher minor sbt version ?

Comment (leobenkel) [reply build.sbt:3]: above https://github.com/leobenkel/Soteria/blob/cab958acf9724639429053d9cc572a83052d1797/project/build.properties#L1 ?

**Comment by rleibman** [comment]
> Don't forget to raise `VERSION` number too

ok, but I often see VERSION moved only by maintainers as they accumulate more than one PR into a version.

File: build.sbt (hunk 1 above)
Comment (leobenkel) [reply build.sbt:3]: seem to be `1.2.8` https://www.lightbend.com/blog/scala-sbt-128-patchnotes

Comment (rleibman) [reply build.sbt:3]: Yeah, but it doesn't fix that problem :(

Comment (rleibman) [reply build.sbt:3]: On top of that... when I moved to 1.2.8, for some reason it's no longer including commons-logging, so ParentTest.scala doesn't compile anymore.

Comment (leobenkel) [reply build.sbt:3]: ok

**Comment by leobenkel** [comment]
unit test are failing

**Comment by leobenkel** [comment]
We are now using Stewart to keep everything up to date. If you have time, some of those have small issues that could be fixed :) @rleibman Thanks !

**Pull request closed without merging**